### PR TITLE
Am 1004 master

### PR DESCRIPTION
**Pull request by vijayrajagopalan-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-1005

### Change description ###
Terraform upgrade changes. Note: The branch number am-1004-master is incorrect. Should have been named am-1005-master.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: terraform.tf [hunk 1]
```diff
@@ -0,0 +1,16 @@
+provider "azurerm" {
```
Comment (nitishsingla1234) [inline terraform.tf:1]: do we need to specify azurem provider version?

Comment (nitishsingla1234) [reply terraform.tf:1]: Got it 
version = "~> 2.25"

**Review by nitishsingla1234 — APPROVED** [review]

File: key-vault.tf [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+  name                        = join("-", [var.product, var.env])
+  product                     = var.product
+  env                         = var.env
+  tenant_id                   = var.tenant_id
+  object_id                   = var.jenkins_AAD_objectId
+  resource_group_name         = azurerm_resource_group.rg.name
+  product_group_object_id     = var.product_group_object_id
+  managed_identity_object_id  = var.managed_identity_object_id
+  common_tags                 = local.common_tags
 }
```
Comment (shashank-rastogi) [inline key-vault.tf:12]: @vijayrajagopalan-hmcts  dont u need to set create_managed_identity = true??

Comment (vijayrajagopalan-hmcts) [reply key-vault.tf:12]: Yes @shashank-rastogi. The new MI will be created by the Key Vault module.

**Review by shashank-rastogi — APPROVED** [review]

**Pull request merged**